DylibBinder/StreamWriterExtensions.cs

```diff
@@ -0,0 +1,887 @@
+						if (escapedType.Contains ("T1"))
+							genericCheckerList.Add ("T1");
+						sw.WriteLineWithIndent ($"<parameter index=\"{i}\" publicName=\"{parameters [i].Item1}\" privateName=\"{privateName}\" type=\"{escapedType}\" isVariadic=\"{isVariadic}\"/>");
+					} catch (Exception e) {
+						Console.WriteLine ($"Problem Parsing the type: {parameters [i].Item2} with exception {e.Message}");
+					}
+				}
+			}
+			Exdent ();
+			sw.WriteLineWithIndent ($"</parameterlist>");
+
+			Exdent ();
+			sw.WriteLineWithIndent ($"</parameterlists>");
+			return genericCheckerList;
+		}
+
+		static void WriteFilteredParameters (this StreamWriter sw, string parameterType, string isVariadic, bool isGetter)
+		{
+			sw.WriteLineWithIndent ($"<parameterlists>");
+			Indent ();
+			sw.WriteLineWithIndent ($"<parameterlist index=\"0\">");
+			Indent ();
+
+			sw.WriteGetSetParameter (parameterType, isVariadic, isGetter);
+
+			Exdent ();
+			sw.WriteLineWithIndent ($"</parameterlist>");
+
+			Exdent ();
+			sw.WriteLineWithIndent ($"</parameterlists>");
+		}
+
+		static List<string> WriteInnerX (this StreamWriter sw, ClassContents c, Dictionary<string, List<ClassContents>> innerXDictionary, int depth)
+		{
+			var name = c.Name.ToString ();
+			if (!innerXDictionary.ContainsKey (name))
+				return null;
+
+			var genericsAccumulator = new List<string> ();
+			var innerXClassContents = innerXDictionary [name];
+			foreach (var classContent in innerXClassContents) {
+				var typeString = classContent.TypeDescriptor.Class.EntityKind.ToString ();
+
+				if (typeString != "Class" && typeString != "Struct" && typeString != "Enum")
+					return null;
+
+				typeString = typeString.ToLower ();
+				sw.WriteBasicOpener ($"inner{typeString}");
+
+				switch (typeString) {
+				case "class":
+					var classGenericsLists = sw.WriteClassContents (classContent, "class", innerXDictionary, depth + 1);
+					genericsAccumulator.AddRange (classGenericsLists.Item1);
+					genericsAccumulator.AddRange (classGenericsLists.Item2);
+					break;
+				case "struct":
+					var structGenericsLists = sw.WriteClassContents (classContent, "struct", innerXDictionary, depth + 1);
+					genericsAccumulator.AddRange (structGenericsLists.Item1);
+					genericsAccumulator.AddRange (structGenericsLists.Item2);
+					break;
+				case "enum":
+					var enumGenericsLists = sw.WriteClassContents (classContent, "enum", innerXDictionary, depth + 1);
+					genericsAccumulator.AddRange (enumGenericsLists.Item1);
+					genericsAccumulator.AddRange (enumGenericsLists.Item2);
+					break;
+
+				}
+				sw.WriteBasicCloser ($"inner{typeString}");
+			}
+			return genericsAccumulator;
+		}
+
+		static void WriteGetSetParameter (this StreamWriter sw, string parameterType, string isVariadic, bool isGetter)
+		{
+			if (isGetter)
+				sw.WriteLineWithIndent ($"<parameter index=\"0\" publicName=\"_\" privateName=\"privateName\" type=\"{parameterType}\" isVariadic=\"{isVariadic}\"/>");
```

Comment:
Getters shouldn't have parameters, right?

Reply:
I believe you are correct!